### DISCUSSION TITLE
count is not working as expected under version 2.0.1

### DISCUSSION THREAD

**Discussion by rotkaepchen** [catfan/Medoo#972] [General]
Hi,
if I try to use the count function I got an error.
I use the same code as under version 1.7. 
I have renamed the class to Medoo2 and also the namespace to Medoo2

The where array:

> $where=array('site'=>'xxx');

My errror:

> SELECT COUNT(*) FROM "xxxx" WHERE "site" = 'xxxxxx'
> Fatal error: Uncaught TypeError: Return value of Medoo2\Medoo2::aggregate() must be of the type string or null, int returned in /../classes/medoo2/Medoo2.class.php:1944
> Stack trace:
> #0 /../classes/medoo2/Medoo2.class.php(1959): Medoo2\Medoo2->aggregate('COUNT', 'view_union_curr...', Array, '*', NULL)
> #1 /../classes/Nsre_cpobject.class.php(108): Medoo2\Medoo2->count('view_union_curr...', Array)
> #2 /../classes/Nsre_cpobject.class.php(61): Nsre_cpobject->fill_objects()
> #3 /../xx.php(100): Nsre_cpobject->get_objects()
> #4 {main}
>   thrown in /../classes/medoo2/Medoo2.class.php on line 1944

Thanks :-)

**Comment by catfan** [comment]
What your actual code is using?

We can't reproduce the error.

And open an issue not discussions if have an error case next time.

**Reply by rotkaepchen** [reply]
Hi,
I use medoo version 2.0.1 but this issue also exists for me in version 2.0.0.
My php version:

> PHP 7.3.20 (cli) (built: Dec  1 2020 17:08:48) ( NTS )
> Copyright (c) 1997-2018 The PHP Group
> Zend Engine v3.3.20, Copyright (c) 1998-2018 Zend Technologies
>     with Zend OPcache v7.3.20, Copyright (c) 1999-2018, by Zend Technologies

Database:
> PostgreSQL 11.5 on x86_64-pc-linux-gnu, compiled by gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-36), 64-bit

**Reply by catfan** [reply]
I'm asking you the code you are using on your project.....

How do you call `$database->count()` and what's the value for the parameter?

**Reply by rotkaepchen** [reply]
> $medoonadb2= new Medoo('database_type' => 'pgsql',
>                 'database' => 'xxxx',
>                 'host' => remoteIP,
>                 'username' => USER,
>                 'password' => PASS,
>                 'error' => PDO::ERRMODE_EXCEPTION
>                 ]);
> 
> $where=array(
>                 'AND'=>['history_adminlan_ruleset.policy'=>$policy ,
>                 'history_adminlan_ruleset.site'=>$site,
>                 'nsre_findings.site'=>$site,
>                 'nsre_findings.policy'=>$policy,
>                 'nsre_riskmatrix.risklevel'=>$key,
>                 'nsre_findings.nsrereportid'=>$nsrereportid]
>             );
> 
> $join=array("[<]history_adminlan_ruleset"=>["ruleid"=>"ruleid"],"[<]nsre_riskmatrix"=>["riskid"=>"riskid"] );
> 
> $q = $medoonadb2->count('nsre_findings',$join,"*",$where);
> 
> The same code works with Medoo Version 1.7.
> $q = $medoonadb->count('nsre_findings',$join,"*",$where);

Error:

> Fatal error: Uncaught TypeError: Return value of medoo2\Medoo2::aggregate() must be of the type string or null, int returned in /NADB/www/nadb/classes/medoo2/Medoo2.class.php:1944 Stack trace: #0 /NADB/www/nadb/classes/medoo2/Medoo2.class.php(1959): medoo2\Medoo2->aggregate('COUNT', 'nsre_findings', Array, '*', Array) #1 /NADB/www/nadb/content/admindt/test.php(45): medoo2\Medoo2->count('nsre_findings', Array, '*', Array) #2 {main} thrown in /NADB/www/nadb/classes/medoo2/Medoo2.class.php on line 1944

Also without a join I did have the same issue.

**Comment by catfan** [comment]
OK, I tested that this is the problem about the return value type is different on pgsql.

It fixed on b97423f4a5c21a5bc07652e97cbefaa42865ed99.

**Reply by rotkaepchen** [reply]
Hi catfan,
it is working now.
Thanks for support.
